### DISCUSSION TITLE
How to map additional metadata fields in JSONLD @context.

### DISCUSSION THREAD

**Discussion by surchs** [dandi/helpdesk#23] [Q&A] [answered]
I am trying to write a (JSONLD) query for information in the [`dataStandard` field of `AssetsSummary`](https://github.com/dandi/dandischema/blob/master/dandischema/models.py#L623) but noticed that `dataStandard` is not in the [released `@context`](https://github.com/dandi/schema/blob/master/releases/0.5.1/context.json). As far as I understand, this context is generated by [`generate_context`](https://github.com/dandi/dandischema/blob/master/dandischema/metadata.py#L20) which seems to [skip fields that don't have a `nskey` attribute set](https://github.com/dandi/dandischema/blob/master/dandischema/metadata.py#L64). The `dataStandard` field does not have a `nskey`.

I think `dataStandard` could fit with the schema.org `encodingFormat` property (https://schema.org/encodingFormat). My questions are:
- is there a reason `dataStandard` is currently not in the context
- how could I contribute a suggested new mapping (e.g. to the `sdo:encodingFormat`), i.e. where would this information go?
- would you like me to do a PR for that?

**Comment by satra** [comment, marked as answer]
@surchs - a PR would be much welcome. for this perhaps we keep `dataStandard`, we do use `encodingFormat` for the actual files being used, while standards were meant to refer to a more meta organizational standard and could eventually include multiple. 

the missing nskey was an oversight, so a PR would be good perhaps use `nskey=dandi` for now, unless you can think of a better property.

**Reply by surchs** [reply]
Ah, great. Thanks @satra, that makes sense. And I didn't realize that the attribute names are equivalent to their mapped vocabulary terms (plus the namespace). I'll make a PR for the `dandi:dataStandard` mapping.

I guess a follow-up question would be: since the context mapping seems to take the form of `current_field_name + name_space_prefix`, how would I be able to map e.g. the `AssetSummary.dataStandard` field to a term with a different name, e.g. `example:dataType`? Would we then change the `dataStandard` field to match the new mapping name?

**Reply by satra** [reply]
there are two ways of doing this. one is in the context file generation itself or building out a separate mapping component to the dandi namespace using `sameAs` or some such avenue. we have not decided on this yet, so any thoughts are welcome.

**Reply by surchs** [reply]
Thanks for the clarification, I'll give it a little think myself!
